index.js

```diff
@@ -477,6 +477,9 @@ module.exports = function (config, deps) {
         .end(
         function (err, res) {
           if (err != null) {
+            if (err.status !== 201) {
```

Comment:
Let's get @darinkrauss 's opinion. To my knowledge `createDatasetForUser` is only used by uploader, so I don't see any side effects by changing this to what uploader needs in terms of useful error info.